tensorpipe/test/channel/channel_test.h

```diff
@@ -202,6 +203,62 @@ class ChannelTestCase {
 
 template <typename TBuffer>
 class ClientServerChannelTestCase : public ChannelTestCase<TBuffer> {
+  std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
+  accept(
+      std::shared_ptr<tensorpipe::transport::Listener> listener,
```

Comment:
Nit: the listener is only needed to issue the `accept` call, hence this function doesn't need a _copy_ of the shared_ptr that holds the listener, it could simply take a reference (in fact, a const reference). However, in turn, it would be odd to take a reference to a (smart) pointer (imagine seeing `void foo(uint8_t* const& data)`), it would make more sense to get a reference to the value pointed to by the pointer, rather than the pointer itself. Hence, to sum up, I think the most "idiomatic" way to pass this in here is `tensorpipe::transport::Listener& listener`. This has the benefit that it doesn't mandate how the lifetime of the listener is managed (could be a shared_ptr, a unique_ptr, stack-allocated, ...).

I'm being somewhat pedantic about it because I had spent some time thinking about it and wondering what's the best approach, and wanted to shared my thoughts. This is also explained much better than I could do here: https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/